### PR TITLE
Add initial state for Production ready Theia with DevWorkspace

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
This is an initial state for Production-ready Theia with DevWorkspace enhancement.

We don't have set enhancement template, but if we will find it useful, we'll come to some.
Now please comment in PR changes/push directly or propose your changes against `theiaDevWorkspace` branch.

### CODE REVIEW

File: enhancements/devworkspace-theia/theia-devworkspace.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+Here are question which should be addressed somehow:
+
+- How Theia based workspace is created with factory flow?
+  - Dashboard after resolving Devfile, determinate the custom Editor, is noone is detected, Dashboard with help of Theia Library resolves editor/plugins DevWorkspaceTemplates with help of .vscode/extensions.json and .che/che-theia-plugins.yaml.
+  Details in [dashboard-creation-diagram.png](./images/dashboard-workspace-creation.png). Source https://docs.google.com/drawings/d/1BJeQMHu6uDP4gqCVU2kA_voCqgb9KaCmJjAevtc9R78/edit?usp=sharing
+- How Theia plugins are defined in DevWorkspace format?
+  - Pure VSX: that's an attribute on Che Theia DevWorkspace Template
+  - Sidecar current:
+    - Plugin's devfile v2 is merged with users devfile and stored together in DevWorkspace CR. Depending on the sidecar policy, user's first container may be modified (provision env vars, endpoints, override entrypoint) or a separate container is provisioned.      
+  - Sidecar alternative: that's devfile v2 served by Che Plugin Registry which should live in a separate DevWorkspace Template
```
Comment (sleshchenko) [inline enhancements/devworkspace-theia/theia-devworkspace.md:23]: @benoitf @l0rd
Sidecar policy appeared instantly for me, I wonder if you considered this alternative where plugins live in their DWT(DevWorkspaceTemplate) and we don't use dev container to simplify flow.
Then later we'll switch to a single container approach, and hopefully, we'll find a straightforward format to make flow clear.

**Comment by azatsarynnyy** [comment]
What is `enhancement template`?

File: enhancements/devworkspace-theia/theia-devworkspace.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+This issue is about finalizing Che Theia workflow to be production ready for DevWorkspace Engine.
+
+## Motivation
+
+We know that in long-term perspective we would like to move away from Theia in VSCode direction to use resources we spent on supporting Theia on making Che flow better.
+But we need in a short time finalize DevWorkspace support to be based on devfile v2 (which is already used by ODO and OpenShift Console).
+
+## Design
+
+Here are question which should be addressed somehow:
```
Comment (benoitf) [inline enhancements/devworkspace-theia/theia-devworkspace.md:14]: It's really strange, it's saying that question need to be addressed but they're all answered.

File: enhancements/devworkspace-theia/theia-devworkspace.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# Production ready Theia DevWorkspace
+
+## Summary
+
+This issue is about finalizing Che Theia workflow to be production ready for DevWorkspace Engine.
```
Comment (azatsarynnyy) [inline enhancements/devworkspace-theia/theia-devworkspace.md:5]: > This issue is about ...

Is there a missed link for some issue?

File: enhancements/devworkspace-theia/theia-devworkspace.md (hunk 2 above)
Comment (sleshchenko) [reply enhancements/devworkspace-theia/theia-devworkspace.md:14]: their answers are proposal, which is not final decisions but I agree that format may be more clearer

**Pull request closed without merging**